app/assets/javascripts/components/autocomplete.js

```diff
@@ -133,5 +135,60 @@ window.GOVUK.Modules = window.GOVUK.Modules || {};
     $select.parentNode.removeChild($select)
   }
 
+  Autocomplete.prototype.initAutoCompleteSearchTopics = function () {
+    var millerColumns = document.querySelector('miller-columns')
+    var topics = millerColumns.taxonomy.flattenedTopics
+    var topicBreadcrumbs = millerColumns.taxonomy.allTopicNames
+
+    var topicSuggestions = []
+
+    topicBreadcrumbs.forEach((breadcrumbs, index) => {
+      topicSuggestions.push({
+        index: index,
+        topic: breadcrumbs[breadcrumbs.length - 1].trim(),
+        breadcrumbs: breadcrumbs
+      })
+    })
+
+    new window.accessibleAutocomplete({ // eslint-disable-line no-new, new-cap
+      id: 'topics-autocomplete',
+      name: 'topics-autocomplete',
+      element: this.$module,
+      minLength: 3,
+      autoselect: false,
+      source: function (query, syncResults) {
+        var results = topicSuggestions
+
+        syncResults(query
+          ? results.filter(function (result) {
+            var indexOf = result.topic.toLowerCase().indexOf(query.toLowerCase())
+            var resultContainsQuery = indexOf !== -1
+            if (resultContainsQuery) {
+              // Strip existing <mark> tags
+              result.breadcrumbs[result.breadcrumbs.length - 1] = result.breadcrumbs[result.breadcrumbs.length - 1].replace(/<\/?mark>/gm, '')
+              // Wrap query in <mark> tags
+              var queryRegex = new RegExp('(' + query + ')', 'ig')
+              result.breadcrumbs[result.breadcrumbs.length - 1] = result.breadcrumbs[result.breadcrumbs.length - 1].replace(queryRegex, '<mark>$1</mark>')
+            }
+            return resultContainsQuery
+          }) : []
+        )
+      },
+      templates: {
+        inputValue: function (result) {
+          return result && ''
+        },
+        suggestion: function (result) {
+          return result && result.breadcrumbs.join(' › ')
+        }
+      },
+      onConfirm: function (result) {
+        if (result && result.index && !topics[result.index].selected) {
+          millerColumns.taxonomy.topicClicked(topics[result.index])
+        }
+      }
```

Comment:
Ahh ok I see how the allTopicNames fits into this. Can I make a counter suggestion, which fits into how I was pondering this.

syncResults -> runs filter on allTopics and checks topic name, returns an array of topics
template methods -> convert topics into appropriate string
onConfirm -> result is a topic so no need to use index

Does that sound workable? I could be missing a key part where this crashes the autocomplete of course

Reply:
I'll give it a try

Reply:
I tried the suggestions. It remove the duplicated source and simplifies the code overall, but requires a bit of logic for the suggestions, as we don't want to edit the topics themselves, so we work with a copy of `topicNames` to do the changes.

Reply:
Ah great! I'd suggest rather than modifying the object you return an object like 

`{ topic: result, highlightedTopicName: result.topicName.replace(queryRegex, '<mark>$1</mark>' }`

(or could even extend that to create the template string in advance since it's basically doing that under a different key)

then you could avoid having to remove stuff but would have to do result.topic in other places.

Reply:
I would avoid creating a new array of objects for every typed letter, especially since there will be a lot of repetition in terms of matched results.

Reply:
Updated as discussed, to re-create the `topicSuggestions` source and operate changes over it.